### PR TITLE
Mechs check if unlocked when unlinking

### PR DESCRIPTION
**Pull request by Afya212**
Added a child lock so you can no longer lock yourself out of your mech.
May or may not have done this before.

Mechs must be in maintenance mode to unlink.

### CODE REVIEW

**Comment by Afya212** [comment]
!review

**Review by Ben10083 — CHANGES_REQUESTED** [review]

**Review by Ben10083 — CHANGES_REQUESTED** [review]
This SHOULD allow the exo to say why it won't unlink, however do we really want hatch to be unlocked? If maintenance mode is on, they can just set a new leader.

File: code/modules/heavy_vehicle/mech_interaction.dm [hunk 1]
```diff
@@ -667,6 +667,9 @@
 			
 			// unlink the leader to get a new one
```
Comment (Ben10083) [inline code/modules/heavy_vehicle/mech_interaction.dm:668]:
```suggestion
			// unlink the leader to get a new one
			if(findtext(text, "unlink"))
				if(hatch_locked) // Can't lock yourself out
					say("Hatch locked, cannot unlink.")
					return
				if(!maintenance_protocols)
					say("Maintenance protocols must be engaged to unlink.")
					return
```

**Review by Ben10083 — CHANGES_REQUESTED** [review]
Should work but make sure to check to ensure I properly overriden the old code with this.

**Review by Ben10083 — CHANGES_REQUESTED** [review]
Needs some feedback to say why it doesn't unlink.

**Review by Ben10083 — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**